pkg/asset/internal/templates.go

```diff
@@ -51,6 +51,9 @@ spec:
         - --kubeconfig=/etc/kubernetes/kubeconfig
         - --require-kubeconfig
         - --lock-file=/var/run/lock/kubelet.lock
+        {{ with .CloudProvider -}}
+        - --cloud-provider={{ . }}
+        {{ end -}}
```

Comment:
We should these changes from the self-hosted kubelet manifest. Otherwise the self-hosted kubelet could behave differently than the on-host kubelet (as documented). The kubelet should use "auto-detect" for both -- so no change should be necessary.